tests/test_notify.py

```diff
@@ -567,6 +567,116 @@ def callback(report):
             "type": "userCallbackSetSeverity"
         })
 
+    def test_middleware_stack_order_legacy(self):
+        def first_callback(notification):
+            notification.meta_data['test']['array'].append(1)
+
+        def second_callback(notification):
+            notification.meta_data['test']['array'].append(2)
+
+        # Simulate an internal middleware
+        bugsnag.legacy.configuration.internal_middleware.before_notify(
+                first_callback)
+        # Add a regular callback function
+        bugsnag.before_notify(second_callback)
```

Comment:
Should you swap these so `bugsnag.before_notify` comes before `bugsnag.configuration.internal_middleware.before_notify` to ensure that internal middleware is still called first?